src/app/dashboard/events/page.tsx

```diff
@@ -1,17 +1,113 @@
+    },
+  });
+  if (!response.ok) {
+    return (
+      <div className="flex w-full flex-col items-center gap-4">
+        <Alert variant="destructive">
+          <AlertCircle className="size-6" />
+          <AlertTitle>Wystąpił błąd podczas pobierania danych.</AlertTitle>
+          <AlertDescription>
+            Sprawdź swoje połączenie z internetem i spróbuj ponownie.
+          </AlertDescription>
+        </Alert>
+      </div>
+    );
+  }
+  const events = (await response.json()) as Event[];
   return (
     <div className="flex flex-col gap-4">
-      <h1 className="text-3xl font-bold">Wszystkie wydarzenia - 0</h1>
-      <Button className="max-w-fit" asChild>
-        {/* TODO display list of events */}
-        <Link href="/dashboard/events/2">
-          Kliknij by przejść do widoku zarządzania eventem
-        </Link>
-      </Button>
+      <div className="space-y-8">
+        <div className="flex justify-between">
+          <div>
+            <h1 className="text-3xl font-bold">Moje wydarzenia</h1>
+            <h3 className="text-muted-foreground">{events.length} wydarzeń</h3>
+          </div>
+          <CreateEventForm />
+        </div>
+        <div className="grid grid-cols-1 gap-4 sm:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 2xl:grid-cols-5">
+          {events.map((event) => (
+            <div key={event.id}>
+              <div className="relative">
+                <Image
+                  src={
+                    event.photoUrl == null
+                      ? EventPhotoPlaceholder
+                      : `${PHOTO_URL}/${event.photoUrl}`
+                  }
+                  width="500"
+                  height="500"
+                  className="aspect-square w-full rounded-t-xl object-cover"
+                  alt={event.name}
+                />
+                <div className="absolute inset-0 z-10 flex h-full flex-col justify-between p-4">
+                  <div className="flex flex-row justify-between">
+                    <EventInfoDiv className="bg-accent-foreground/60 text-background">
+                      <Calendar1 size={16} />
+                      <p className="text-sm">
+                        {format(event.startDate, "dd.MM.yyyy HH:mm")}
+                      </p>
+                    </EventInfoDiv>
+                    <EventInfoDiv className="bg-accent-foreground/60 text-background">
+                      <p className="text-sm">{event.participantsCount}</p>
+                      <Users size={16} />
+                    </EventInfoDiv>
+                  </div>
+                  {/* Mockup suggested that there also should be location,
+                   * but for now the backend does not return it - mejsiejdev
+                   */}
+                </div>
+              </div>
+              <div className="flex flex-col gap-4 rounded-b-xl border-x border-b border-muted-foreground p-4">
+                <p className="text-2xl font-bold">{event.name}</p>
```

Comment:
Nazwa eventu też może być bardzo długa
```suggestion
                <p className="line-clamp-3 text-2xl font-bold">{event.name}</p>
```